docs/DEVELOPMENT.md

```diff
@@ -86,149 +93,87 @@ Wait until the migration completes and logs to show
+
+Use `http://localhost:8080` as the `Homepage URL` and `Authorization callback URL`.
+
+After creation, add the OAuth Client ID as `GH_CLIENT_ID` and Client Secret as `GH_CLIENT_SECRET` in [.env.dev][env-dev].
+
+For `JWT_SIGNING_KEY`, you can use any random word.
+
 ### Running API Service
 
 Once the database is setup and the migration has been run, you can run api service by
 
 ```bash
-go run ./cmd/hub
+go run ./cmd/api
 ```
 
 ### Running tests
 
 To run the tests, we need a test db.
-
 - If you have installed postgresql, create a `hub_test` database.
 - If you are running a container, create `hub_test` database in the same container.
 
-  ```bash
+```bash
   source .env.dev
 
   docker exec -it hub bash -c \
     "PGPASSWORD=$POSTGRES_PASSWORD \
      psql -h localhost -p 5432 -U postgres -c 'create database hub_test;'"
-  ```
-
-Once the `hub_test` database is created, you can run the test using following command:
-
-```bash
-  go test  -count=1 -v ./...
-```
-
-**NOTE:** `tests` use the database configurations from [test/config/env.test][env-test-file]
-
----
-
-## Running On Kubernetes Cluster
-
-### Prerequisites:
-
-- Kubernetes Cluster:
-  - You can run a cluster on your local machine using [CRC][install-crc], [Minikube][install-minikube] or [kind][install-kind]
-
-- Depending on the cluster you have, install the command line tool:
-
-  - [`kubectl`][install-kubectl]: For interacting with cluster.
-  - [`oc`][install-oc]: For interacting with OpenShift cluster.
-  - [`ko`][install-ko]: tool for building and deploying Golang applications to Kubernetes.
-
-### Deploying API and DB Service
-
-Export `KO_DOCKER_REPO` for `ko` to publish image to. E.g.
-
-```bash
-export KO_DOCKER_REPO=quay.io/<username>
 ```
 
-Log into the registry used in `KO_DOCKER_REPO` so that `ko` gets
-access to push the image to the registry.
-
-Make sure you are logged into the cluster and you are in the
-`hub/api` directory before running the following command
+Once the `hub_test` database is created, you can run the test using following command:
 
 ```bash
-ko apply -f config/
-```
-
-The command above will create an image and push it to registry pointed by  `KO_DOCKER_REPO` and then deploy the `api` and `db` container on the cluster.
-
-**NOTE:** Ensure that the image is **publicly** available otherwise deployment will fail. The pod status will give `Error: ImagePullBackOff`
-
-Watch the pods until both of them are running.
-```
-kubectl get pods -o wide -w
-```
-
-To create tables and to populate, we need to run the migration. Run the following command:
-
-```
-ko apply -f config/db-migration/14-db-migration.yaml
+  go test -p 1 -count=1 -v ./pkg/...
 ```
 
-The command above will create a migration image, push to registry pointed by `KO_DOCKER_REPO` and run a `job` that performs the migration.
-
-Check the logs using
-
+To re-generate the golden files use the below command
 ```bash
-kubectl logs job/db-migration
+go test ./pkg/... -test.update-golden
```

Comment:
This test is failing on my machine :thinking: . Getting 
```
flag provided but not defined: -test.update-golden
```

Reply:
command fails because some package doesn't have `testdata` dir as they don't use golden file for test.. so we need to run it on only those pkg which uses golden file.
```
go test $(go list -f '{{ .ImportPath }} {{ .TestImports }}' ./... | grep gotest.tools/v3/golden | awk '{print $1}' | tr '\n' ' ') -test.update-golden=true
```
this would fix that :smile: